src/analysis/normed_space/star/multiplier.lean

```diff
@@ -0,0 +1,446 @@
+
+section scalars
+
+variables {S : Type*} [monoid S] [distrib_mul_action S A] [smul_comm_class 𝕜 S A]
+  [has_continuous_const_smul S A] [is_scalar_tower S A A] [smul_comm_class S A A]
+
+instance : has_smul S 𝓜(𝕜, A) :=
+{ smul := λ s a,
+  { to_prod := s • a.to_prod,
+    central := λ x y, show (s • a.snd) x * y = x * (s • a.fst) y,
+      by simp only [continuous_linear_map.smul_apply, mul_smul_comm, smul_mul_assoc, central] } }
+
+@[simp] lemma smul_fst (k : 𝕜) (a : 𝓜(𝕜, A)) : (k • a).fst = k • a.fst := rfl
+@[simp] lemma smul_snd (k : 𝕜) (a : 𝓜(𝕜, A)) : (k • a).snd = k • a.snd := rfl
+
+end scalars
+
+instance : has_one 𝓜(𝕜, A) := ⟨⟨1, λ x y, rfl⟩⟩
+
+instance : has_mul 𝓜(𝕜, A) :=
+{ mul := λ a b,
+  { to_prod := (a.fst.comp b.fst, b.snd.comp a.snd),
+    central := λ x y, show b.snd (a.snd x) * y = x * a.fst (b.fst y),
+      by simp only [central] } }
+
+instance : has_nat_cast 𝓜(𝕜, A) :=
+{ nat_cast := λ n, ⟨n, λ x y,
+  begin
+    rw [prod.snd_nat_cast, prod.fst_nat_cast],
+    simp only [←nat.smul_one_eq_coe, smul_apply, one_apply, mul_smul_comm, smul_mul_assoc],
+  end⟩ }
+
+instance : has_int_cast 𝓜(𝕜, A) :=
+{ int_cast := λ n, ⟨n, λ x y,
+  begin
+    rw [prod.snd_int_cast, prod.fst_int_cast],
+    simp only [←int.smul_one_eq_coe, smul_apply, one_apply, mul_smul_comm, smul_mul_assoc],
+  end⟩ }
+
+instance : has_pow 𝓜(𝕜, A) ℕ :=
+{ pow := λ a n, ⟨a.to_prod ^ n, λ x y,
+  begin
+    induction n with k hk generalizing x y,
+    { refl },
+    { rw [prod.pow_snd, prod.pow_fst] at hk ⊢,
+      rw [pow_succ a.snd, mul_apply, a.central, hk, pow_succ' a.fst, mul_apply] },
+  end⟩ }
+
+@[simp] lemma add_to_prod (a b : 𝓜(𝕜, A)) : (a + b).to_prod = a.to_prod + b.to_prod := rfl
+@[simp] lemma zero_to_prod : (0 : 𝓜(𝕜, A)).to_prod = 0 := rfl
+@[simp] lemma neg_to_prod (a : 𝓜(𝕜, A)) : (-a).to_prod = -a.to_prod := rfl
+@[simp] lemma sub_to_prod (a b : 𝓜(𝕜, A)) : (a - b).to_prod = a.to_prod - b.to_prod := rfl
+@[simp] lemma one_to_prod : (1 : 𝓜(𝕜, A)).to_prod = 1 := rfl
+@[simp] lemma nat_cast_to_prod (n : ℕ) : (n : 𝓜(𝕜 , A)).to_prod = n := rfl
+@[simp] lemma int_cast_to_prod (n : ℤ) : (n : 𝓜(𝕜 , A)).to_prod = n := rfl
+@[simp] lemma pow_to_prod (n : ℕ) (a : 𝓜(𝕜, A)) : (a ^ n).to_prod = a.to_prod ^ n := rfl
+
+lemma add_fst (a b : 𝓜(𝕜, A)) : (a + b).fst = a.fst + b.fst := rfl
+lemma add_snd (a b : 𝓜(𝕜, A)) : (a + b).snd = a.snd + b.snd := rfl
+lemma zero_fst : (0 : 𝓜(𝕜, A)).fst = 0 := rfl
+lemma zero_snd : (0 : 𝓜(𝕜, A)).snd = 0 := rfl
+lemma neg_fst (a : 𝓜(𝕜, A)) : (-a).fst = -a.fst := rfl
+lemma neg_snd (a : 𝓜(𝕜, A)) : (-a).snd = -a.snd := rfl
+lemma sub_fst (a b : 𝓜(𝕜, A)) : (a - b).fst = a.fst - b.fst := rfl
+lemma sub_snd (a b : 𝓜(𝕜, A)) : (a - b).snd = a.snd - b.snd := rfl
+lemma one_fst : (1 : 𝓜(𝕜, A)).fst = 1 := rfl
+lemma one_snd : (1 : 𝓜(𝕜, A)).snd = 1 := rfl
+@[simp] lemma mul_fst (a b : 𝓜(𝕜, A)) : (a * b).fst = a.fst * b.fst := rfl
+@[simp] lemma mul_snd (a b : 𝓜(𝕜, A)) : (a * b).snd = b.snd * a.snd := rfl
```

Comment:
Missing `mul_to_prod`

Reply:
That one doesn't work because of the reverse multiplication in the second coordinate.